sdk/core/azure-core/src/main/java/com/azure/core/util/UrlBuilder.java

```diff
@@ -140,7 +142,25 @@ public String getPath() {
      * @return The provided query parameter name and encoded value to query string for the final URL.
      */
     public UrlBuilder setQueryParameter(String queryParameterName, String queryParameterEncodedValue) {
-        query.put(queryParameterName, queryParameterEncodedValue);
+        query.put(queryParameterName, new QueryParameter(queryParameterName, queryParameterEncodedValue));
```

Comment:
Does using `QueryParameter` now introduce a potential NPE that didn't exist before if `queryParameterName` is null? Not something that needs to change but would be worth documenting.

Reply:
Solved in the appendQueryParameter method as well.